From: Jan Engelhardt <ej@inai.de>
Date: 2025-02-09 21:28:54.517217439 +0100

Fix some terrible code.

[   28s] /home/abuild/rpmbuild/BUILD/grpc-1.70.1-build/grpc-1.70.1/src/core/lib/transport/call_state.h:157:3: error: control reaches end of non-void function [-Werror=return-type]
---
 src/core/call/call_state.h                  |    5 +++++
 src/core/channelz/channelz.h                |    1 +
 src/core/lib/promise/inter_activity_mutex.h |    2 ++
 3 files changed, 8 insertions(+)

Index: grpc-1.74.0/src/core/call/call_state.h
===================================================================
--- grpc-1.74.0.orig/src/core/call/call_state.h
+++ grpc-1.74.0/src/core/call/call_state.h
@@ -155,6 +155,7 @@ class CallState {
       case ClientToServerPullState::kTerminated:
         return "Terminated";
     }
+    return "Undefined";
   }
   template <typename Sink>
   friend void AbslStringify(Sink& out, ClientToServerPullState state) {
@@ -185,6 +186,7 @@ class CallState {
       case ClientToServerPushState::kFinished:
         return "Finished";
     }
+    return "Undefined";
   }
   template <typename Sink>
   friend void AbslStringify(Sink& out, ClientToServerPushState state) {
@@ -235,6 +237,7 @@ class CallState {
       case ServerToClientPullState::kTerminated:
         return "Terminated";
     }
+    return "Undefined";
   }
   template <typename Sink>
   friend void AbslStringify(Sink& out, ServerToClientPullState state) {
@@ -275,6 +278,7 @@ class CallState {
       case ServerToClientPushState::kFinished:
         return "Finished";
     }
+    return "Undefined";
   }
   template <typename Sink>
   friend void AbslStringify(Sink& out, ServerToClientPushState state) {
@@ -305,6 +309,7 @@ class CallState {
       case ServerTrailingMetadataState::kPulledCancel:
         return "PulledCancel";
     }
+    return "Undefined";
   }
   template <typename Sink>
   friend void AbslStringify(Sink& out, ServerTrailingMetadataState state) {
Index: grpc-1.74.0/src/core/channelz/channelz.h
===================================================================
--- grpc-1.74.0.orig/src/core/channelz/channelz.h
+++ grpc-1.74.0/src/core/channelz/channelz.h
@@ -144,6 +144,7 @@ class BaseNode : public DualRefCounted<B
       case EntityType::kCall:
         return "call";
     }
+    return "unknown";
   }
 
   static std::optional<EntityType> KindToEntityType(absl::string_view kind) {
Index: grpc-1.74.0/src/core/lib/promise/inter_activity_mutex.h
===================================================================
--- grpc-1.74.0.orig/src/core/lib/promise/inter_activity_mutex.h
+++ grpc-1.74.0/src/core/lib/promise/inter_activity_mutex.h
@@ -307,6 +307,8 @@ class InterActivityMutex {
         case State::kMovedFrom:
           LOG(FATAL) << "Mutex acquirer already moved from";
       }
+      // Control never actually reaches here
+      return Pending{};
     }
 
    private:
